### PR TITLE
Fix slippage compenstation

### PR DESCRIPTION
**Pull request by numberZero**
Current calculation sets `elapsed` to a negative value, thus making delay greater than `interval` while it should be less (by `fn` execution time approximately).

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -136,13 +136,13 @@ function Timer:is_active()
 	return self.active == true
 end
 
-function Timer:expire(elapsed)
+function Timer:expire()
```
Comment (sofar) [inline init.lua]: Unrelated change, make a separate commit, or, separate PR please.

File: timerdemo/init.lua [hunk 2]
```diff
@@ -11,11 +11,11 @@ assert(s)
 
 -- our timer functions
 local mytimerfunc1 = function(elapsed)
-	print("fn1: ", elapsed)
+	print("fn1: ", minetest.get_us_time(), elapsed)
 end
 
 local mytimerfunc2 = function(elapsed)
-	print("fn2: ", elapsed)
+	print("fn2: ", minetest.get_us_time(), elapsed)
```
Comment (sofar) [inline timerdemo/init.lua:18]: Ok, yes, this is fine, and probably useful to folks.

Comment (sofar) [reply timerdemo/init.lua:18]: missed the anonymous function, do this one too.

File: init.lua (hunk 1 above)
Comment (sofar) [reply init.lua]: needs API doc updates.

Comment (numberZero) [reply init.lua]: #2

File: timerdemo/init.lua (hunk 2 above)
Comment (numberZero) [reply timerdemo/init.lua:18]: Fixed

**Comment by numberZero** [comment]
Note that there are no safety measures for the case of server overload, like tick skipping in MC.
